### PR TITLE
chore!: alter predicate signature

### PR DESCRIPTION
**Pull request by petertonysmith94**
# Breaking Changes

- Changed `Predicate` class constructor parameters from `inputData` to `data`

```ts
// Before
const predicate = new Predicate({
  ...unchangedParameters,
  inputData,
})
```

```ts
// After
const predicate = new Predicate({
  ...unchangedParameters,
  data,
})
```

- Type generated Predicate now accepts a single parameter for initialization 

```ts
// Before
PredicateTrueAbi__factory.createInstance(provider, data, configurableConstants);
```

```ts
// After
new PredicateTrue({
  provider,
  data,
  configurableConstants
});
```

# Checklist

- [X] I **_added_** — `tests` to prove my changes
- [ ] I **_updated_** — all the necessary `docs`
- [X] I **_reviewed_** — the entire PR myself, using the GitHub UI
- [ ] I **_described_** — all breaking changes and the Migration Guide

### CODE REVIEW

File: packages/abi-typegen/src/templates/predicate/main.hbs [hunk 1]
```diff
@@ -47,27 +47,32 @@ export type {{capitalizedName}}Configurables = Partial<{
...
 const bytecode = '{{hexlifiedBinString}}';
 
-export class {{capitalizedName}} extends Predicate<{{capitalizedName}}Inputs> {
+export class {{capitalizedName}} extends Predicate<{{capitalizedName}}Inputs, {{capitalizedName}}Configurables> {
 
   static readonly abi = abi;
   static readonly bytecode = bytecode;
 
-  constructor(provider: Provider, predicateData?: {{capitalizedName}}Inputs{{#if configurables}}, configurables?: {{capitalizedName}}Configurables{{/if}}) {
+  constructor(provider: Provider, params?: {{capitalizedName}}Parameters) {
```
[the remark this answers is not in the document]
Comment (petertonysmith94) [reply packages/abi-typegen/src/templates/predicate/main.hbs:69]: It looks ascetically pleasing, however, matching `Predicate` class seems wise.

File: packages/abi-typegen/test/fixtures/templates/predicate-with-configurable/main.hbs [hunk 2]
```diff
@@ -95,18 +101,18 @@ const abi = {
...
+  constructor(params: MyPredicateParameters) {
     super({
       abi,
       bytecode,
-      provider,
-      inputData: predicateData,
-      configurableConstants: configurables,
+      provider: params.provider,
+      data: params?.data,
+      configurableConstants: params?.configurableConstants,
```
Comment (arboleya) [inline packages/abi-typegen/test/fixtures/templates/predicate-with-configurable/main.hbs:115]: Do we need the `?`, considering `params` are not optional?

```suggestion
      data: params.data,
      configurableConstants: params.configurableConstants,
```

File: packages/abi-typegen/test/fixtures/templates/predicate/main.hbs [hunk 3]
```diff
@@ -293,17 +299,18 @@ const abi = {
...
-  constructor(provider: Provider, predicateData?: MyPredicateInputs) {
+  constructor(params: MyPredicateParameters) {
     super({
       abi,
       bytecode,
-      provider,
-      inputData: predicateData,
+      provider: params.provider,
+      data: params?.data,
+      configurableConstants: params?.configurableConstants,
```
Comment (arboleya) [inline packages/abi-typegen/test/fixtures/templates/predicate/main.hbs:313]: [Same](https://github.com/FuelLabs/fuels-ts/pull/2876/files#r1700556898).

File: packages/abi-typegen/test/fixtures/templates/predicate-with-configurable/main.hbs (hunk 2 above)
Comment (petertonysmith94) [reply packages/abi-typegen/test/fixtures/templates/predicate-with-configurable/main.hbs]: c51c54715898ef288df16082a0ec70ae4563eb87

File: packages/abi-typegen/test/fixtures/templates/predicate/main.hbs (hunk 3 above)
Comment (petertonysmith94) [reply packages/abi-typegen/test/fixtures/templates/predicate/main.hbs]: c51c54715898ef288df16082a0ec70ae4563eb87

**Review by arboleya — APPROVED** [review]
Awesome! Many thanks on this one! 🚢

**Review by danielbate — APPROVED** [review]

**Pull request merged**